src/hooks/useMaciSignup.ts

```diff
@@ -8,10 +8,19 @@ import { api } from "~/utils/api";
 import { useEthersSigner } from "./useEthersSigner";
 import { useAccount } from "wagmi";
 
+export enum MaciSignUpError {
+  NO_KEY = "No Maci Key",
+  NO_ATTESTATION = "Don't Have Attestation",
+  NO_SIGNER = "No Signer",
+  UNKNOWN = "Unknown",
+}
+
 export interface IUseMaciSignUpData {
   stateIndex?: string;
   isRegistered?: boolean;
   onSignup: () => Promise<void>;
+  error?: MaciSignUpError;
+  setError: (error?: MaciSignUpError) => void
```

Comment:
Why do you need to expose this state setter?

Reply:
actually I just wanna call **set error to `undefined`**, I should add a function instead of exposing the state setter :)